src/functions.php

```diff
@@ -213,6 +213,37 @@ function capture(
     return trim($process->getOutput());
 }
 
+/**
+ * @param string|array<string|\Stringable|int>       $command
+ * @param array<string, string|\Stringable|int>|null $environment
+ */
+function get_exit_code(
+    string|array $command,
+    array|null $environment = null,
+    string|null $path = null,
+    float|null $timeout = null,
+    bool|null $quiet = null,
+    Context $context = null,
+): int {
+    $process = run(
+        command: $command,
+        environment: $environment,
+        path: $path,
+        timeout: $timeout,
+        allowFailure: true,
+        context: $context,
+        quiet: $quiet,
+    );
+
+    $code = $process->getExitCode();
+
+    if ($code === null) {
+        throw new \LogicException('The process has not been terminated.');
```

Comment:
How is it possible?

Reply:
Don't know but this is what the function signatures specifies

```
     * @return int|null The exit status code, null if the Process is not terminated```

Reply:
okay, but we know the process is finished... but okay.

Reply:
I know, but I want to return an `int` not a nullable int. for better DX. What should I do in this case?

Reply:
`return $process->getExitCode()` But I understand you added this code for phpstan. Not a big deal